### PR TITLE
Add "How it Works" section to readme

### PR DESCRIPTION
**Pull request by spalger**

### CODE REVIEW

**Comment by silverwind** [comment]
:+1:

File: README.md [hunk 1]
```diff
@@ -70,3 +70,8 @@ Exports
 * **Server** - A class similar to https.Server (except instances have `setTimeout()` from http.Server).
 
 * **createServer**(< _object_ >tlsConfig[, < _function_ >requestListener]) - _Server_ - Creates and returns a new Server instance.
+
+How it Works
+============
+
+TLS and HTTP connections are easy to distiguish based on the first byte sent by clients trying to connect. See [this comment](https://github.com/mscdex/httpolyglot/issues/3#issuecomment-173680155) for more informatoin.
```
Comment (mscdex) [inline README.md]: s/distiguish/distinguish
s/informatoin/information

**Comment by mscdex** [comment]
A couple of nits but otherwise LGTM

File: README.md (hunk 1 above)
Comment (spalger) [inline README.md]: woah :) editing on github doesn't have spell checking!

**Comment by spalger** [comment]
fixed the typos, thanks @mscdex

**Comment by mscdex** [comment]
Landed in ef54b90ac5. Thanks!

**Pull request closed without merging**